pkgs/servers/nosql/keydb/default.nix

```diff
@@ -0,0 +1,60 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, libuuid, curl
+, openssl
+, pkg-config
+, systemd
+}:
+
+stdenv.mkDerivation rec {
+  pname = "keydb";
+  version = "6.3.1";
+
+  src = fetchFromGitHub {
+    owner = "snapchat";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-00Dx6GRAXVpJUXhu4kjsVCEsoN615vhGvc7+gViERqI=";
+  };
+
+  postPatch = ''
+    substituteInPlace deps/lua/src/Makefile \
+      --replace "ar rcu" "${stdenv.cc.targetPrefix}ar rcu"
+    substituteInPlace src/Makefile \
+      --replace "as --64 -g" "${stdenv.cc.targetPrefix}as --64 -g"
+  '';
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+  buildInputs = [
+    (curl.override { inherit openssl; })
+    libuuid
+    systemd
+    openssl
+  ];
+
+  makeFlags = [
+    "PREFIX=${placeholder "out"}"
+    "USE_SYSTEMD=yes"
+    "BUILD_TLS=yes"
```

Comment:
It would be nice to have easy way of disabling TLS, maybe you can add `buildTLS ? true` in derivation arguments.

Reply:
Why? Doesn't seem very useful to me tbh.

Reply:
Then the openssl dependency can be optional, only added when TLS is enabled and maybe libcurl also.

Reply:
Why is that a desirable feature? Those libraries will be on the system anyways.